From: SeongJae Park <sj@kernel.org>
Date: Sat, 2 Aug 2025 10:28:09 -0700
Subject: [PATCH] mm/damon: introduce damon_operations_attrs for operations set
 control

It is not scalable to make new operations set for every extension.
Introduce a new struct for customizations of operations set.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 11 +++++++++++
 mm/damon/core.c       |  1 +
 2 files changed, 12 insertions(+)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index c35ed89371d0..09e6e176c582 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -655,6 +655,16 @@ struct damon_operations {
 	void (*cleanup)(struct damon_ctx *context);
 };
 
+/**
+ *
+ * struct damon_operations_attrs - Monitoring operations control attributes.
+ *
+ * @use_reports:	Whether to use damon_report_access()-ed information.
+ */
+struct damon_operations_attrs {
+	bool use_reports;
+};
+
 /*
  * struct damon_call_control - Control damon_call().
  *
@@ -814,6 +824,7 @@ struct damon_ctx {
 	struct mutex kdamond_lock;
 
 	struct damon_operations ops;
+	struct damon_operations_attrs ops_attrs;
 
 	struct list_head adaptive_targets;
 	struct list_head schemes;
diff --git a/mm/damon/core.c b/mm/damon/core.c
index 428ac1b83118..c47b900ef9d6 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1221,6 +1221,7 @@ int damon_commit_ctx(struct damon_ctx *dst, struct damon_ctx *src)
 	if (err)
 		return err;
 	dst->ops = src->ops;
+	dst->ops_attrs = src->ops_attrs;
 
 	return 0;
 }
-- 
2.39.5

